algorithms/src/std_algorithms/impl/Kokkos_FindEnd.hpp

```diff
@@ -80,24 +80,28 @@ struct StdFindEndFunctor {
         m_p(std::move(p)) {}
 };
 
+//
+// exespace impl
+//
 template <class ExecutionSpace, class IteratorType1, class IteratorType2,
           class BinaryPredicateType>
-IteratorType1 find_end_impl(const std::string& label, const ExecutionSpace& ex,
-                            IteratorType1 first, IteratorType1 last,
-                            IteratorType2 s_first, IteratorType2 s_last,
-                            const BinaryPredicateType& pred) {
+IteratorType1 find_end_exespace_impl(const std::string& label,
+                                     const ExecutionSpace& ex,
+                                     IteratorType1 first, IteratorType1 last,
+                                     IteratorType2 s_first,
+                                     IteratorType2 s_last,
+                                     const BinaryPredicateType& pred) {
   // checks
   Impl::static_assert_random_access_and_accessible(ex, first, s_first);
   Impl::static_assert_iterators_have_matching_difference_type(first, s_first);
   Impl::expect_valid_range(first, last);
   Impl::expect_valid_range(s_first, s_last);
 
   // the target sequence should not be larger than the range [first, last)
-  namespace KE            = ::Kokkos::Experimental;
-  const auto num_elements = KE::distance(first, last);
-  const auto s_count      = KE::distance(s_first, s_last);
+  namespace KE                        = ::Kokkos::Experimental;
+  const auto num_elements             = KE::distance(first, last);
+  [[maybe_unused]] const auto s_count = KE::distance(s_first, s_last);
```

Comment:
Isn't this always used (in line 115)?

Reply:
the complain comes from line 106, it sees this possibility of existing earlier

Reply:
the same should be true for num_elements though , but don't recall seeing that warning

Reply:
I will try to remove them again

Reply:
Let me know.